lib/ui/search/search.dart

```diff
@@ -28,6 +32,17 @@ class _SearchState extends State<Search> {
 
     _searchFocusNode = FocusNode();
     _searchController = TextEditingController();
+    if (widget.searchTerm != null) {
+      bloc.search(SearchTermEvent(widget.searchTerm));
+      _searchController.text = widget.searchTerm;
+    }
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    // hide soft keyboard if search value is passed from discovery view
+    if (widget.searchTerm != null) FocusScope.of(context).requestFocus(FocusNode());
```

Comment:
can we remove this code and move this logic to when the search is clicked?

Reply:
No, I don't think we can. They are unrelated, we already have a remove focus line on clicking search button.

Reply:
What we can do is edit autofocus of TextField

`autofocus: widget.searchTerm != null ? false : true,`

Reply:
@erdemyerebasmaz sounds like a good solution to me.